### PR TITLE
prepare for Mastodon in the federation stats and include BlaBlaNet

### PR DESCRIPTION
**Pull request by tobiasd**

### CODE REVIEW

**Comment by annando** [comment]
What exactly is "BlaBlaNet"? Isn't it just some manipulated version code thing?

**Comment by tobiasd** [comment]
Their github page says it's an enhanced fork from Hubzilla. On My node they make up a significant portion from the Hubzilla numbers, so I added them to the displayed platforms.

**Comment by rabuzarus** [comment]
this is their github repos https://github.com/BlaBlaNet/BlaBlaNet-Social-Network But there is nothing interesting inside

(????)

**Comment by annando** [comment]
Concerning Mastodon servers: They don't reveal their version numbers for security reasons: https://github.com/Gargron/mastodon/issues/141

If they don't even return that they are using Mastodon, then we don't have to add them here, I guess.

**Comment by tobiasd** [comment]
Bummer :-( Well then there could still be a counter for how many Mastodon nodes are known. I really like to show the diversity of the network the node is part of on that page.

**Comment by annando** [comment]
When I find a reliable way to detect that it is Mastodon then we can do it.

**Comment by annando** [comment]
We will now be able to detect mastodon. How do we detect "BlaBlaNet"?

**Comment by tobiasd** [comment]
I have several of them already in the DB. Approx 1/3 of the Hubzilla nodes my node knows.

**Comment by annando** [comment]
BTW: I just saw that I'm having a huge amount of Hubzilla servers in my list without any version number. I guess this has to be dealt in your statistics scripts as well?

**Comment by tobiasd** [comment]
So far they are intentionally excluded. But, as we don't get the version from Mastodon, that exclusion has to be dropped anyhow right?

**Comment by annando** [comment]
I think so. We should add these servers as well. Do you want to add it to this pull request?

**Comment by tobiasd** [comment]
Yes; but, withouth the version number the simple notation of `(40)` looks strange, so I want to add something like `unknown version` before the brackets

File: mod/admin.php [hunk 1]
```diff
@@ -269,34 +269,47 @@ function admin_page_federation(&$a) {
...
+		// to the version string for the displayed list.
+		$newV = array();
+		foreach ($v as $vv) {
+			if ($vv['version'] == '') {
+				$newV[] = array('total'=>$vv['total'], 'version'=>t('unknown'));
+			} else {
+				$newV[] = $vv;
+			}
+		}
+ 		$v = $newV;
```
Comment (Hypolite) [inline mod/admin.php]: If you want to modify an array in a `foreach`, here's a more common approach that don't require copying the entire array:
````php
foreach ($array as $key => $value) {
  if ($condition) {
    $array[$key] = $modified_value;
  }
}

Comment (tobiasd) [inline mod/admin.php]: I wont change the other foreach loops into that syntax style though; at least not at the moment. It took me alsmost an hour to wrap my mind around it and several wrong attemps. Maybe at some rainy day when I have nothing other on my todo-list ;-)

**Pull request merged**